x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -710,6 +712,27 @@ private void handleIndexActionAuthorizationResult(
                 )
             );
         } else {
+            if (request instanceof IndicesRequest.Replaceable replaceable) {
+                var indexExpressions = replaceable.getResolvedIndexExpressions();
+                if (indexExpressions != null) {
+                    indexExpressions.expressions().forEach(resolved -> {
+                        if (resolved.localExpressions().localIndexResolutionResult() == CONCRETE_RESOURCE_UNAUTHORIZED) {
+                            resolved.localExpressions()
+                                .setException(
+                                    actionDenied(
+                                        authentication,
+                                        authzInfo,
+                                        action,
+                                        request,
+                                        IndexAuthorizationResult.getFailureDescription(List.of(resolved.original()), restrictedIndices),
+                                        null
+                                    )
+                                );
+                        }
+                    });
+                }
+            }
```

Comment:
Having it here works. But it feels a bit out of place in this method. Instead, can we extract it as a method and invoke it [here](https://github.com/elastic/elasticsearch/blob/466309e8e81b15a49c03aaeb6b200b7bf71ff760/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java#L535-L544) before the call to `handleIndexActionAuthorizationResult`?